presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayMinMaxUtils.java

```diff
@@ -92,7 +93,10 @@ public static Double doubleArrayMinMax(MethodHandle compareMethodHandle, Type el
                     return null;
                 }
                 double value = elementType.getDouble(block, i);
-                if ((boolean) compareMethodHandle.invokeExact(value, selectedValue)) {
+                if (isNaN(value)) {
+                    return value;
```

Comment:
nit: somehow returning `NaN` looks better to me

Reply:
Oops, I think so. I will update it.